torchsynth/synth.py

```diff
@@ -236,6 +239,34 @@ def set_hyperparameter(self, hyperparameter: Tuple[str, str, str], value: Any):
         assert not ModuleParameter.is_parameter_frozen(parameter)
         setattr(parameter.parameter_range, hyperparameter[2], value)
 
+    def save_hyperparameters(self, filename: str) -> None:
+        """
+        Save hyperparameters to a JSON file
+        """
+        hp = [{"name": key, "value": val} for key, val in self.hyperparameters.items()]
+        with open(os.path.abspath(filename), "w") as fp:
+            json.dump(hp, fp, indent=True)
+
+    def load_hyperparameters(self, name: str) -> None:
+        """
+        Load hyperparameters from a JSON file
+        """
+
+        # Try to load nebulae from package resources, otherwise, try
+        # to load from a filename
+        try:
+            synth = type(self).__name__.lower()
+            nebulae_str = f"nebulae/{synth}/{name}.json"
```

Comment:
idea here is that we can add nebula to the package for each synth and that will exist in that package and you can just load it by calling the nebula name for that synth -- like "default" or "drum" and it will load that from the package resources. But, you can also just load nebula from a json file where ever. Doesn't have to be apart of the package.